tests/TestCase/Http/ClientTest.php

```diff
@@ -898,4 +898,96 @@ public function testRedirectDifferentSubDomains()
         $this->assertInstanceOf(Response::class, $result);
         $this->assertSame($response, $result);
     }
+
+    /**
+     * Scheme is set when passed to client in string
+     */
+    public function testCreateScopedClientSetsProtocol()
+    {
+        $client = Client::createScopedClientFromUrl('https://example.co/');
+        $this->assertSame('https', $client->getConfig('scheme'));
+    }
+
+    /**
+     * Host is set when passed to client in string
+     */
+    public function testCreateScopedClientSetsHost()
+    {
+        $client = Client::createScopedClientFromUrl('https://example.co/');
+        $this->assertSame('example.co', $client->getConfig('host'));
+    }
+
+    /**
+     * Test exception is thrown when URL cannot be parsed
+     */
+    public function testCreateScopedClientThrowsInvalidExceptionWhenUrlCannotBeParsed()
+    {
+        try {
+            Client::createScopedClientFromUrl('htps://');
+        } catch (InvalidArgumentException $e) {
+            $this->assertInstanceOf(InvalidArgumentException::class, $e);
+            $this->assertTextContains('did not parse', $e->getMessage());
+
+            return;
+        }
+        $this->fail('InvalidArgumentException was not thrown when string was not parsed');
+    }
+
+    /**
+     * Port is set when passed to client in string
+     */
+    public function testCreateScopedClientSetsPort()
+    {
+        $client = Client::createScopedClientFromUrl('https://example.co:8765/');
+        $this->assertSame(8765, $client->getConfig('port'));
+    }
+
+    /**
+     * Test exception is throw when no scheme is provided.
+     */
+    public function testCreateScopedClientThrowsInvalidArgumentExceptionWhenNoSchemeProvided()
+    {
+        try {
+            Client::createScopedClientFromUrl('example.co');
+        } catch (InvalidArgumentException $e) {
```

Comment:
You can use `expectException()` instead of try/catch.

Reply:
Corrected